src/test/java/org/jenkinsci/plugins/workflow/test/steps/SemaphoreStep.java

```diff
@@ -234,6 +234,23 @@ public static class Execution extends AbstractStepExecutionImpl {
             }
         }
 
+        @Override public void onResume() {
```

Comment:
I added an `onResume` implementation for `semaphore` so that it can be used with `@LocalData`. I don't think it should make a difference for tests that use `JenkinsSessionRule`, although maybe we should guard this logic and do nothing if the state is not empty. Let me see if we have decent test coverage for `semaphore` here.

Reply:
I think I have generally used a real step in conjunction with `@LocalData`; feels a bit uneasy to serialize a `SemaphoreStep.Execution` in persistent test data.

Reply:
Fair enough. I switched to `sleep` in [b5e199f](https://github.com/jenkinsci/workflow-support-plugin/pull/263/commits/b5e199fde4eb418111d162a5a28402d05ff4caa2).